src/core/border_router/routing_manager.cpp

```diff
@@ -641,26 +641,48 @@ void RoutingManager::EvaluateNat64Prefix(void)
 
     NetworkData::Iterator            iterator = NetworkData::kIteratorInit;
     NetworkData::ExternalRouteConfig config;
-    bool                             hasAdvertisedNat64Prefix = false;
+    Ip6::Prefix                      smallestNat64Prefix;
 
     otLogInfoBr("Evaluating NAT64 prefix");
 
+    smallestNat64Prefix.Clear();
     while (Get<NetworkData::Leader>().GetNextExternalRoute(iterator, config) == kErrorNone)
     {
-        if (config.mNat64 && config.GetPrefix().IsValidNat64())
+        const Ip6::Prefix &prefix = config.GetPrefix();
+        if (config.mNat64 && prefix.IsValidNat64())
         {
-            hasAdvertisedNat64Prefix = true;
+            if (smallestNat64Prefix.GetLength() == 0 || prefix < smallestNat64Prefix)
+            {
+                smallestNat64Prefix.Clear();
+                smallestNat64Prefix = prefix;
+            }
         }
     }
 
-    if (!hasAdvertisedNat64Prefix && !mIsAdvertisingLocalNat64Prefix)
+    if (smallestNat64Prefix.GetLength() == 0 || smallestNat64Prefix == mLocalNat64Prefix)
     {
-        // TODO: when multiple prefixes exits, remove the ones with lower preference or numerically larger.
-        if (AddExternalRoute(mLocalNat64Prefix, NetworkData::kRoutePreferenceLow, /* aNat64= */ true) == kErrorNone)
+        otLogInfoBr("No valid NAT64 prefix or the local NAT64 prefix is the smallest one in Thread network");
+
+        // Advertise local NAT64 prefix.
+        if (!mIsAdvertisingLocalNat64Prefix &&
+            AddExternalRoute(mLocalNat64Prefix, NetworkData::kRoutePreferenceLow, /* aNat64= */ true) == kErrorNone)
         {
             mIsAdvertisingLocalNat64Prefix = true;
         }
     }
+    else
+    {
+        // Withdraw local NAT64 prefix if it's not the smallest one in Network Data.
+        // TODO: remove the prefix with lower preference after discovering upstream NAT64 prefix is supported
+        if (mIsAdvertisingLocalNat64Prefix && smallestNat64Prefix < mLocalNat64Prefix)
+        {
+            otLogInfoBr("Withdrawing local NAT64 prefix since a smaller one %s exists in the Thread network",
+                        smallestNat64Prefix.ToString().AsCString());
```

Comment:
```suggestion
            otLogNoteBr("Withdrawing local NAT64 prefix since a smaller one %s exists",
                        smallestNat64Prefix.ToString().AsCString());
```
This event seems important and rare.

Reply:
Done.